pkg/3scale/amp/operator/upgrade.go

```diff
@@ -758,6 +767,107 @@ func (u *UpgradeApiManager) findDeploymentTriggerOnImageChange(triggerPolicies [
 	return result, nil
 }
 
+func (u *UpgradeApiManager) upgradePodTemplateLabels() (reconcile.Result, error) {
+	apicast, err := Apicast(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	backend, err := Backend(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	zync, err := Zync(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	memcached, err := Memcached(u.apiManager)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	system, err := System(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	deploymentConfigs := []*appsv1.DeploymentConfig{
+		apicast.StagingDeploymentConfig(),
+		apicast.ProductionDeploymentConfig(),
+		backend.ListenerDeploymentConfig(),
+		backend.WorkerDeploymentConfig(),
+		backend.CronDeploymentConfig(),
+		zync.DeploymentConfig(),
+		zync.QueDeploymentConfig(),
+		memcached.DeploymentConfig(),
+		system.AppDeploymentConfig(),
+		system.SidekiqDeploymentConfig(),
+		system.SphinxDeploymentConfig(),
+	}
+
+	if !u.apiManager.IsExternalDatabaseEnabled() {
+		redis, err := Redis(u.apiManager, u.Client())
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		deploymentConfigs = append(deploymentConfigs, redis.SystemDeploymentConfig())
+		deploymentConfigs = append(deploymentConfigs, redis.BackendDeploymentConfig())
+	}
+
+	if u.apiManager.IsSystemPostgreSQLEnabled() {
+		systemPostgreSQL, err := SystemPostgreSQL(u.apiManager, u.Client())
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		deploymentConfigs = append(deploymentConfigs, systemPostgreSQL.DeploymentConfig())
+	}
+
+	if u.apiManager.IsSystemMysqlEnabled() {
+		systemMySQL, err := SystemMySQL(u.apiManager, u.Client())
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		deploymentConfigs = append(deploymentConfigs, systemMySQL.DeploymentConfig())
+	}
+
+	if !u.apiManager.IsZyncExternalDatabaseEnabled() {
+		deploymentConfigs = append(deploymentConfigs, zync.DatabaseDeploymentConfig())
+	}
+
+	updated := false
+	for _, desired := range deploymentConfigs {
+		updatedTmp, err := u.ensurePodTemplateLabels(desired)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		updated = updated || updatedTmp
+	}
+
+	return reconcile.Result{Requeue: updated}, nil
+}
+
+func (u *UpgradeApiManager) ensurePodTemplateLabels(desired *appsv1.DeploymentConfig) (bool, error) {
```

Comment:
With this changes here we will ensure new labels are added, but the old labels will still be there. Should we clean them? too?

Reply:
Added procedure to remove old labels

Reply:
This method will need to be reused in future 3scale versions, without the part that removes the old metering labels. If you want you can add a comment to remember to delete that part

Reply:
done